addons/mail/models/discuss/discuss_channel.py

```diff
@@ -313,7 +313,10 @@ def _get_from_context_or_raise(self, thread_id):
         """Overridden because guests and (portal) users need sudo to post on channels. This ensures they are actually
         members before granting them access, as well as properly setting up the guest context on the resulting thread
         if applicable."""
-        return self.env["discuss.channel.member"]._get_as_sudo_from_context_or_raise(thread_id).channel_id
+        channel_sudo = self.env["discuss.channel.member"]._get_as_sudo_from_context_or_raise(thread_id).channel_id
+        if not channel_sudo:
+            raise NotFound()
```

Comment:
Not necessary to make diff in this file, if the channel.member is found, the channel is guaranteed to exist (required field).